### ISSUE TITLE
Add a contents pane option to the MooseDocs navigation extension

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#27360]
### Motivation

It can be difficult to navigate some longer pages in MooseDocs and be able to see the overall page organization with the different heading levels, leading to some confusion.

### Design

This would add a toggle to MooseDoc's `NavigationExtension` called `contentspane` or something (false by default), and then a user could enable it on a page like
```
!config navigation contentspane=True

(some page content)
```
This would produce a left pane with hyperlinks that looks like:
```
Food
  Fruits
    Apple
    Banana
  Vegetables
    Carrots
```

### Impact

Improved readability of documentation pages.

**Comment by cticenhour** [comment]
Can you elaborate on how this would be different from `!config navigation scrollspy=True`? This is the current default for general markdown pages, and the main limitation is the level of links that are shown (only the second level [`##`] if memory serves). If that is the main addition you want to make, I would say let's explore how to enhance `scrollspy` to include the lower heading levels too.

**Comment by joshuahansel** [comment]
Yeah, I think a general number of levels and maybe also the option to have it in the same column as the text instead of a right sidebar, where I think it would be difficult to fit all of the levels.
